utils/logalloc.cc

```diff
@@ -788,28 +788,36 @@ class mmap_memory_segment_store_backend : public segment_store_backend {
     }
 };
 
-#ifndef SEASTAR_DEFAULT_ALLOCATOR
 class segment_store {
-    memory::memory_layout _layout;
-    uintptr_t _segments_base; // The address of the first segment
+    std::unique_ptr<segment_store_backend> _backend;
+
+private:
+    static std::unique_ptr<segment_store_backend> make_backend() {
+#ifndef SEASTAR_DEFAULT_ALLOCATOR
+        return std::make_unique<seastar_memory_segment_store_backend>();
+#else
+        // We want to simulate 3GB mem per shard (1 << 30).
```

Comment:
3GB = 3 << 30

Reply:
and why 3GB?

Reply:
I guess just a typo. For tools 1GB is likely too much, but it's not a real problem.

Reply:
I copied the comment and calculation from the existing std memory backed segment store: https://github.com/scylladb/scylladb/blob/fe401f14de638668ce539f589da9e6b8bb71814b/utils/logalloc.cc#L731

Reply:
Actually, yes its 1GB.

Reply:
Note that we just `mmap()` this memory but don't `mlock` it, so if no one uses it, the kernel doesn't have to materialize the backing. I will still use less for tools, will leave 1GB as the default (for tests).

Reply:
Although lsa will allocate all segments it can on startup, but doesn't write to them. It is a good question whether this will trigger assigning physical memory to these pages.

Reply:
Why do we use mmap? This will reduce the ability of asan to track it.

Reply:
IIRC the code has support for poison/unpoison already.